kernel/mmu.c

```diff
@@ -380,7 +380,12 @@ static void free_page_frame_list_put(struct z_page_frame *pf)
 {
 	PF_ASSERT(pf, z_page_frame_is_available(pf),
 		 "unavailable page put on free list");
-	sys_slist_append(&free_page_frame_list, &pf->node);
+	/* TODO: Check why this assertion hits */
+	/*PF_ASSERT(pf, IS_PTR_ALIGNED(pf, void*), "unaligned pf pointer");*/
+	/* The structure is packed, which ensures that this is true */
+	void *node = (void *)pf;
```

Comment:
I don't think the explicit typecast is needed. Isn't one of the main feature of void pointers that you don't need to do explicit casts to/from them.

Reply:
Absolutely, will check and remove. Thanks!

Reply:
Should be done now @jhedberg, thanks for catching this.

Reply:
I've also removed them from the inverse assignment (void to a data type) since those are not necessary either.